src/ReplicatedStorage/Client/InstanceComponents/Misc/CameraLook.lua

```diff
@@ -0,0 +1,164 @@
+local Players = game:GetService("Players")
+local RunService = game:GetService("RunService")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local TweenService = game:GetService("TweenService")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+
+local CameraLook = {}
+CameraLook.__index = CameraLook
+
+local function calculateJointAngle(lookVector, sourcePosition, targetPosition)
+	return math.asin((sourcePosition - targetPosition):Dot(lookVector)/(sourcePosition - targetPosition).Magnitude)
+end
+
+local easingStyle = Enum.EasingStyle.Circular
+local easingDirection = Enum.EasingDirection.In
+local function dampenAngle(sourceAngle: number, goalAngle: number, angularSpeed: number, deltaTime: number)
+	local angleDiff = goalAngle%math.rad(360) - sourceAngle%math.rad(360)
+	if math.abs(angleDiff) > math.rad(180) then
+		local dividend = math.sign(angleDiff)*math.rad(180)
+		angleDiff = angleDiff%dividend - dividend
+	end
+	return sourceAngle + math.sign(angleDiff) * TweenService:GetValue(math.clamp(math.min(angularSpeed, math.abs(angleDiff/deltaTime) / angularSpeed), 0, 1), easingStyle, easingDirection) * angularSpeed * deltaTime
+end
+
+function CameraLook.new(instance: Instance)
+	local self = setmetatable({}, CameraLook)
+
+	self.Instance = instance
+	self._trove = Trove.new()
+
+	-- Wait for Humanoid to be added
+	local humanoid = instance:FindFirstChildWhichIsA("Humanoid")
```

Comment:
Perhaps GetHumanoidAsync could be a utility function in ReplicatedStorage/Common/CharacterUtil.lua

Reply:
Would prefer to add this in a different PR down the line

Reply:
But otherwise, I agree